2d birder post

maelle opened the pull request.

maelle commented:
@mstrimas could you have a look? 😸

mstrimas commented:
Looks great, nice to have a full comparison of the two packages! Some points, mostly for your own interest:

1. Yes, names dropped as of last release as a consequence of GDPR, but I see you found the workaround for seeing them 😉. There's also Top 100 lists for countries, [states](https://ebird.org/top100?locInfo.regionCode=DE-BW&year=AAAA), and hotspots
2. Exotic/domestic species are (mostly) not accepted by eBird, which explains the Black Swan case. [There is talk of addressing this](https://help.ebird.org/customer/portal/questions/5390335-exclusion-of-not-native-birds-in-lists), but it is a complicated issue.
3. eBird started in 2002, but only became global in June 2010, which explains the trend you're seeing in Germany. Here's the global trend:
![ebird-growth-2017-11-520x308](https://user-images.githubusercontent.com/7296482/43836046-0cdfa12e-9ae2-11e8-8035-bc1d730312f6.png)
4. Maybe worth adding a brief note about the ability to get presence/absence data with `auk`, this is a major advantage of using eBird data vs. almost all other large open species occurrence data sets.

maelle commented:
Thanks a ton, I'll update my post!

Do I understand correctly that the black swan observations are not considered as _errors_ then? That's what I thought before reading your answer: I thought that it had been established that the birds seen were probably not black swans. 🙈

maelle commented:
In my home region we have this exotic species: https://en.wikipedia.org/wiki/African_sacred_ibis established after a few escaped from a local zoo :sob: and I see it's not only in Africa in eBird's data https://ebird.org/camerica/map/sacibi2?neg=true&env.minX=-72.92348046874997&env.minY=6.713814420238199&env.maxX=-66.46351953124997&env.maxY=9.676672734001569&zh=true&gp=true&ev=Z&mr=1-12&bmo=1&emo=12&yr=all&byr=1900&eyr=2018 🤔

maelle commented:
Direct link to (now updated) post https://deploy-preview-243--ropensci.netlify.com/blog/2018/08/21/birds-radolfzell/

mstrimas commented:
@maelle haha you're wading into a complicated issue that I'm probably not qualified to address. Birders have complicated rules about countability of introduced and exotic species relating to whether they're [established](http://listing.aba.org/criteria-determining-establishment-exotics/), so some of these species are "countable" and others aren't. eBird will keep records of non-countable species that have been rejected; however, they won't show up in most places on the website. A big part of this is distinguishing between legitimate rarities and vagrants, which birders get excited about, and e.g. escaped domestic birds, which they don't. I'll try to ask some of the serious birders around here about this to get a better answer.

sckott commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,520 @@
+
+#### API key? Not yet
+
+At the moment, `rebird` interfaces the 1.1 APIs that will be retired
```
version 1.1

sckott commented on content/blog/2018-08-21-birds-radolfzell.md:
````diff
@@ -0,0 +1,520 @@
+mapr::map_leaflet(birds2)
+```
+
+    ## no. taxa > 9, using single color - consider passing in colors
````
you could set `warnings=FALSE` and possibly `messages=FALSE` for the knitr chunks to avoid thsese

sckott commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,520 @@
+
+#### Preparing the dataset
+
+Here, the worflow is to *clean* the data and to *filter* it using an
```
workflow

sckott replied:
remove `an`

maelle commented:
@mstrimas I added a remark about the renaming of `auk_extent`.

@sckott Thanks, I implemented all changes. I can't figure out how to force the line break before "Quite handy!" after the `mapr` map, any clue?

sckott commented:
did you try a `<br>` for the break?

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+We got 49 observations (`nrow(trimmed_birds)`) of 49 species
+(`length(unique(trimmed_birds$comName))`), over 2 places
+(`length(unique(trimmed_birds$locName))`) during 5 observation sessions.
+Hopefully merely an appetizer to what we can hope to get from using the
```
s/what we can hope to get/what we can get/ 

slightly redundant hopes

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+  geom_sf(data = mpi_sf,
+          shape = 2) +
+  hrbrthemes::theme_ipsum()+
+  ggtitle("eBird observations over the last 30 days",
```
this plot is so similar to above plot of obs in last 30 days that I took a while to see the triangle even when it was in the subtitle. Can you do without showing this map?

maelle replied:
Good idea, I put the MPI on the first map, which makes the code chunk a bit long but I added comments.

I actually think it's quite easy when reading to skip code... I hope I'm not wrong.

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+After getting access to a custom dataset corresponding to the EBD for
+Germany only, I used `auk`’s documentation and [this
+post](https://ropensci.org/blog/2018/08/07/auk/) to learn how to process
+it. Since I wasn’t planning on zero-filling the data, I was able to
```
do you need to explain "zero-filling" as in "zero-filling the data to get X"

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+few other places](https://en.wikipedia.org/wiki/Black_swan) but eBird
+mostly doesn’t accept the entry of exotic species [although it’s
+debated](https://help.ebird.org/customer/portal/questions/5390335-exclusion-of-not-native-birds-in-lists).
+In any case, eBird’s control of the data entered is quite admirable.
```
s/control/curation/ ?

maelle replied:
👌

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+
+#### R packages for occurrence data
+
+In this post we gave a rough view of what birds are present in the
```
You can say "I" in "In this post we gave a rough..."

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+-   `rebird` for getting access to recent data via an API. `rebird` is
+    part of a larger collection of packages for occurrence data within
+    rOpenSci’s suite, with `spocc` being an umbrella package accessing
+    several data sources; `scrubr` an helper for cleaning data obtained
```
s/an helper/a helper/

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+    part of a larger collection of packages for occurrence data within
+    rOpenSci’s suite, with `spocc` being an umbrella package accessing
+    several data sources; `scrubr` an helper for cleaning data obtained
+    this way; and `mapr` an utility package for mapping such data.
```
s/an utility/a utility/

maelle replied:
whyyy? English is so complicated!!

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+    several data sources; `scrubr` an helper for cleaning data obtained
+    this way; and `mapr` an utility package for mapping such data.
+
+Explore these packages, and more rOpenSci’s packages, by checking out
```
too many "packages"

Explore these rOpenSci packages, and more by checking out our packages page!

stefaniebutland commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+#### More birding soon!
+
+Stay tuned for the next post in this series, that’ll mark a break from
+modern data since we’ll try and extract information from old natural
```
s/try and extract/try to extract/

maelle commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+
+### How to access eBird data?
+
+There are two ways to access eBird data, whole dataset download or APIs,
```
Done! All the lists look super bad of course cf https://github.com/ropensci/roweb2/issues/172 😿

maelle commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+want nor need to download the whole eBird Basic Dataset (EBD), one can
+request a custom download, which I did, asking for only the data for
+Germany which I got after a few days (the time to receiving the link to
+download a custum dataset is variable). While waiting, I worked on the
```
it's time I used Jeroen's spelling package on all my PRs 🙈

maelle commented on content/blog/2018-08-21-birds-radolfzell.md:
```diff
@@ -0,0 +1,528 @@
+In the following, we’ll use the rOpenSci’s packages `rebird` to get and
+map all observations in the last 30 days near Radolfzell in Germany.
+
+The *Radolfzell* part of that sentence is a bit different than in last
```
👍

maelle commented:
@sckott `<br>` worked, so weird that I needed it, but thanks a lot for the tip!

@stefaniebutland many thanks, all should be implemented now.
